src/sql/parts/notebook/models/cell.ts

```diff
@@ -230,13 +230,15 @@ export class CellModel implements ICellModel {
 				}
 			}
 		} catch (error) {
+			let message: string;
 			if (error.message === 'Canceled') {
 				// swallow the error
+				message = localize('executionCanceled', 'Query execution is canceled');
+			}else{
+				message = notebookUtils.getErrorMessage(error);
 			}
-			let message = notebookUtils.getErrorMessage(error);
 			this.sendNotification(notificationService, Severity.Error, message);
 			// TODO track error state for the cell
-			throw error;
```

Comment:
Don't you still want to throw the error if it wasn't 'Canceled'?

Reply:
No. 'throw error' used to be in catch block and so there is no way of throwing error message when it wasn't canceled

Reply:
I'm not sure what you mean - it would look like this 
```
} catch (error) {
			let message: string;
			if (error.message === 'Canceled') {
				// swallow the error
				message = localize('executionCanceled', 'Query execution was canceled');
			} else {
				message = notebookUtils.getErrorMessage(error);
			}

                                this.sendNotification(notificationService, Severity.Error, message);
                        if(error.message === 'Canceled') {
                                 throw error;
                        }
```

Reply:
I don't think throwing error is needed in this case since we are notifying the error in the catch back itself. This runCell method is used in three different places and it's a simple action which doesn't have any dependency to pass it on. We don't seem to be handling error in all three parent methods.

Reply:
Sounds good. Might want to move the "swallow error" comment out of the if though since we're swallowing all errors.

Does the error display in any other way? Like in the cell itself? The notification is nice but it might be good to have the grid actually display that an error occurred too.

Reply:
Error displays in all cases whenever there is an exception, however, we are customizing the error message only in case of canceled scenario. runCell method corresponds to each code component in notebook not in a Grid. We will need to handle it in a different way when we implement run all cells. Folks can correct me if it doesn't make sense.